storage/pebble/registers/payload/lookup.go

```diff
@@ -0,0 +1,87 @@
+package payload
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/pebble/registers/config"
+)
+
+// lookupKey is the encoded format of the storage key for looking up register value
+type lookupKey struct {
+	encoded []byte
+}
+
+// newLookupKey takes a height and registerID, returns the key for storing the register value in storage
+func newLookupKey(height uint64, reg flow.RegisterID) *lookupKey {
+	key := lookupKey{
+		encoded: make([]byte, 0, len(reg.Owner)+1+len(reg.Key)+1+config.HeightSuffixLen),
+	}
+
+	// The lookup key used to find most recent value for a register.
+	//
+	// The "<owner>/<key>" part is the register key, which is used as a prefix to filter and iterate
+	// through updated values at different heights, and find the most recent updated value at or below
+	// a certain height.
+	key.encoded = append(key.encoded, []byte(reg.Owner)...)
+	key.encoded = append(key.encoded, '/')
+	key.encoded = append(key.encoded, []byte(reg.Key)...)
+	key.encoded = append(key.encoded, '/')
+
+	// Encode the height getting it to 1s compliment (all bits flipped) and big-endian byte order.
+	// (Prefix iteration in pebble does not support reverse iteration.)
+	onesCompliment := ^height
+	key.encoded = binary.BigEndian.AppendUint64(key.encoded, onesCompliment)
+
+	return &key
+}
+
+// lookupKeyToRegisterID takes a lookup key and decode it into height and RegisterID
+func lookupKeyToRegisterID(lookupKey []byte) (uint64, flow.RegisterID, error) {
```

Comment:
why is this not a method on the lookupKey type? a lot of feature-envy

Reply:
I could move it there, it's a private implementation on RN because of its single usecase